code/Helper/Entity/Producthelper.php

```diff
@@ -139,8 +136,13 @@ public function getProductCollectionQuery($storeId, $productIds = null, $only_vi
                 $catalogInventory_stock->addInStockFilterToCollection($products);
             }
 
-            $products = $products->addAttributeToSelect('special_from_date')->addAttributeToSelect('special_to_date')
-                             ->addAttributeToFilter('status', Mage_Catalog_Model_Product_Status::STATUS_ENABLED);
+            // $products = $products->addFinalPrice();
```

Comment:
Never sure about this one :).

We need to test every combinaison before removing it
- price
- price + special price
- price + group price
- price + catalog rule
- price + group price + special price
- price + catalog rule + special price
- price + catolog rule + group price